src/NHibernate/Impl/CriteriaImpl.cs

```diff
@@ -253,31 +253,27 @@ public ICriteria Add(ICriterion expression)
 
 		public IList List()
 		{
-			var results = new List<object>();
-			List(results);
-			return results;
+			return List<object>().ToIList();
 		}
 
 		public void List(IList results)
```

Comment:
Is this method used?

Reply:
Internally, still used by `CriteriaImpl.Subcriteria.List(IList results)`, since it is implemented by delegation. But indeed `ICriteria.List(IList results)` could now be obsoleted for removal in 6.0, leaving only the generic list method. (There is no `IList List()` method in this interface, contrary to `IQuery` which has `void List(IList results)` and `IList List()`. But on `ISessionImplementor`, there is still a `IList List(CriteriaImpl criteria)`, only referenced by xml comments: maybe it should be obsoleted too.)

By the way, `IQuery.List(IList results)` has also no internal usage. `IList IQuery.List()` has five internal usages, which could all be replaced by calls to `List<object>`. But it has also 908 usages in tests. A similar PR for `IQuery` would be a good thing.

Reply:
Well, there is still a `IList List()` on `ICriteria`, I have missed it the other day.